compiler/src/dotty/tools/dotc/core/Types.scala

```diff
@@ -3469,7 +3455,55 @@ object Types {
     assert(resType.isInstanceOf[TermType], this)
     assert(paramNames.nonEmpty)
 
+    private def setVariances(tparams: List[LambdaParam], vs: List[Variance]): Unit =
+      if tparams.nonEmpty then
+        tparams.head.givenVariance = vs.head
+        setVariances(tparams.tail, vs.tail)
+
+    override val isVariantLambda = variances.nonEmpty
+    if isVariantLambda then setVariances(typeParams, variances)
+
+    def givenVariances =
+      if isVariantLambda then typeParams.map(_.givenVariance)
+      else Nil
+
+    override def computeHash(bs: Binders): Int =
+      doHash(new Binders(this, bs), givenVariances ::: paramNames, resType, paramInfos)
+
+    // No definition of `eql` --> fall back on equals, which calls iso
+
+    final override def iso(that: Any, bs: BinderPairs): Boolean = that match {
+      case that: HKTypeLambda =>
+        paramNames.eqElements(that.paramNames)
+        && isVariantLambda == that.isVariantLambda
+        && (!isVariantLambda
```

Comment:
I think this line and the next two are equivalent to:
```suggestion
        && givenVariances == that.givenVariances
```

Reply:
yes, but it's faster the way it is written. `iso` is performance critical!